### DISCUSSION TITLE
Re-create client preset

### DISCUSSION THREAD

**Discussion by lj-vpd** [dotansimha/graphql-code-generator#8625] [Docs Discussions]
Hello,

Is there an easy/straightforward way to recreate the settings of client preset, but 'manually' (without using `preset: "client"`)?

The reason I ask is that like [this issue](https://github.com/dotansimha/graphql-code-generator/issues/8123), I'm getting duplicates in the output file(s).

However, with the exception of these few duplicates everything else seems perfect (especially the way in which the output results in multiple files such as fragments, gql, graphql.ts, etc).

I'd like to exactly recreate the client preset but with the ability to tweak things in order to resolve the duplicates issue.

**Comment by charlypoly** [comment]
Hi @lj-vpd,

The `client-preset` heavily relies on the [typed-document-node](https://the-guild.dev/graphql/codegen/plugins/typescript/typed-document-node) plugin.

**Reply by lj-vpd** [reply]
Hi @charlypoly – Thank you

I did spot that myself (but thanks for pointing it out)

What I'm struggling to re-create without the `client-preset` is the `fragment-masking.ts` and `gql.ts` – Any pointers there?

**Reply by charlypoly** [reply]
Those features are part of the `client-preset` only and cannot be used independently for now.

Could you detail what you mean by "duplicates"? The issue you are referring to was not using the `client-preset` and had a configuration issue.

Maybe the issue you mention should be fixed in the `client-preset` itself and not by starting a new one.

**Reply by lj-vpd** [reply]
Ahh I see – Understood

The duplicates are mainly of the `Document` types that are generated

I was trying to use both the `client-preset` (in order to get the `fragment-masking.ts` and `gql.ts` files), and after it, some plugins like `typescript-react-apollo` to get hooks (etc)

Together, these resulted in duplicates, so I wanted to replicate `client-preset`, but with more granular control (almost like how one is able to "eject" with Create React App)

**Reply by charlypoly** [reply]
> I was trying to use both the `client-preset` (in order to get the `fragment-masking.ts` and `gql.ts` files), and after it, some plugins like `typescript-react-apollo` to get hooks (etc)

There is no point in using both the `client-preset` and hook-based plugins such as `typescript-react-apollo`.
The `client-preset` allow achieving the same type-safety without having to generate hook wrappers.



> but with more granular control

Why do you mean by that? Maybe we could include new features to `client-preset`

**Reply by lj-vpd** [reply]
> There is no point in using both the client-preset and hook-based plugins such as typescript-react-apollo.
> The client-preset allow achieving the same type-safety without having to generate hook wrappers.

I wanted the convenience and 'cookie cutter' nature of generated hook wrappers, which is basic the root of this duplication issue

> Why do you mean by that? Maybe we could include new features to client-preset

I mean the ability to control the output of the specific plugins that the `client-preset` uses, so that I can for example exclude certain outputs, rename certain outputs (etc)

I basically want the benefits of the `gql` and `fragment-masking` but with more control

**Reply by charlypoly** [reply]
I see, however, maybe a better way to get what you want without forking the preset would be to contribute to the ongoing discussions on allowed `client-preset` options: https://github.com/dotansimha/graphql-code-generator/issues/8562

**Reply by lj-vpd** [reply]
In the ideal world, yes

Unfortunately we're working to a tight internal deadline, so have had to abandon our use of `graphql-code-generator` this time and go back to an old Apollo codegen

Apologies for not being able to be helpful
